credentials/sts/sts.go

```diff
@@ -119,17 +120,16 @@ func NewCredentials(opts Options) (credentials.PerRPCCredentials, error) {
 	if err := validateOptions(opts); err != nil {
 		return nil, err
 	}
-
-	// Load the system roots to validate the certificate presented by the STS
-	// endpoint during the TLS handshake.
-	roots, err := loadSystemCertPool()
-	if err != nil {
-		return nil, err
+	// set a default timeout value
+	// DefaultClient.Timeout=0 is no timeout at all.
+	// Note: this code is added in only because prior implementation of sts.go
+	//  statically set Timeout value of stsRequestTimeout=5s
+	if opts.HTTPClient.Timeout == http.DefaultClient.Timeout {
+		opts.HTTPClient.Timeout = stsRequestTimeout
```

Comment:
This should probably do something like this instead:

```golang
func NewCredentials(opts Options) (credentials.PerRPCCredentials, error) {
	...
	client := opts.HTTPClient
	if opts.HTTPClient == nil {
		client = defaultHTTPClient()
	}

	return &callCreds{
		opts:   opts,
		client: client,
	}, nil
}

func defaultHTTPClient() *http.Client {
	return &http.Client{
		Timeout: stsRequestTimeout,
		Transport: &http.Transport{
			TLSClientConfig: &tls.Config{
				RootCAs: roots,
			},
		},
	}
}
```

Reply:
ii don't think i need to construct a client _and_ specify the systecerts. 

`http.DefaultClient` automatially does that for you. 

in sts.go:

```golang
func NewCredentials(opts Options) (credentials.PerRPCCredentials, error) {
	if err := validateOptions(opts); err != nil {
		return nil, err
	}
	client := opts.HTTPClient
	if opts.HTTPClient == nil {
		client = http.DefaultClient
	}
	return &callCreds{
		opts:   opts,
		client: makeHTTPDoer(client),
	}, nil
}
```

---

i just verified the following construct where the sts server is on cloud run (i.,e its behind a google-issued CA thats in the systemroots

```golang
	stscreds, err := sts.NewCredentials(sts.Options{
		TokenExchangeServiceURI: "https://stsserver-6w42z6vi3q-uc.a.run.app/token",,
		Resource:                *stsaudience,
		Audience:                *stsaudience,
		Scope:                   *scope,
		SubjectTokenPath:        *stsCred,
		SubjectTokenType:        "urn:ietf:params:oauth:token-type:access_token",
		RequestedTokenType:      "urn:ietf:params:oauth:token-type:jwt",
		// HTTPClient:              hc,   /// commeted out so that the defautlClient is used
	})
```